Blog: Best HIPAA compliant AB testing tools (#4138)
PostHog/posthog.com: contents/blog/best-hipaa-compliant-ab-testing-tools.md

These are the broad principles, but **please consult an expert** before making any final decision on how to implement tools in compliance with HIPAA. 

## The best HIPAA-compliant A/B testing tools

### PostHog

![posthog experimentation](../images/blog/hipaa-compliant-ab-testing/posthog.png)

PostHog is an all-in-one platform that includes a [fully-featured experimentation](https://posthog.com/product/experimentation-suite) suite in addition to product analytics, session recording, and a feature flag system that also powers its testing features. It supports both A/B/n and multivariate experiments, which you can target by geography, cohorts, and user properties.

PostHog is open source, which means you can inspect the code yourself and extend the product via custom integrations or apps to satisfy specific edge-case requirements. As an [all-in-one platform](https://posthog.com/product), PostHog makes it easy to create cohorts of users based on user properties, then run targeted experiments on those users alone using feature flags.

#### PostHog and HIPAA compliance

- **Self-Hosting:** Yes
- **BAA Available**: No

> PostHog doesn't currently offer a BAA PostHog Cloud, so using it in compliance with HIPAA means [self-hosting it](/docs/self-host) on your own infrastructure. The [PostHog Marketplace](/marketplace) includes several third-party providers who can support self-hosted deployment, or manage your instance entirely, if you need help.

Author: ```suggestion
PostHog doesn't currently offer a BAA for using PostHog Cloud, so using it in compliance with HIPAA means [self-hosting it](/docs/self-host) on your own infrastructure. The [PostHog Marketplace](/marketplace) includes several third-party providers who can support self-hosted deployment, or manage your instance entirely, if you need help.
```

Reviewer: Technically true - although this may change in the next few weeks as we figure out our own BAA - could we mention that Cloud might be coming soon?

Author: Yeah, we could do something similar to what we did on the GDPR one, potentially.

Author: In the interests of time, I'm going to go live as is and we can update when needed. I don't expect this to pull hue numbers anyway.

### VWO

![vwo testing](../images/blog/hipaa-compliant-ab-testing/vwo-testing.png)

VWO offers a range of different products. Its testing product, VWO Testing, includes visual and code editors. You can run both A/B and multivariate tests, though the latter is limited to its Pro and Enterprise plans.

#### VWO and HIPAA compliance

- **Self-Hosting:** No
- **BAA Available**: Yes

VWO is a cloud-hosted platform only and will enter into a BAA with its users.

What [VWO says](https://vwo.com/compliance/hipaa/):

>"VWO customers who are subject to HIPAA and wish to use the VWO products with PHI must sign a BAA with VWO. Customers are responsible for ensuring that they achieve compliance with HIPAA and HITECH Act requirements. 
>
>"We adhere to the HIPAA obligations by leveraging appropriate security configuration options for all VWO products. Additionally, we make our Business Associate Agreement (BAA) available for execution by subscribers."

#### How much does VWO cost?

VWO doesn't publish its pricing publicly, but it does offer a 14-day free trial.

> > **HIPAA password requirements:** HIPAA requires Covered Entities and Business Associates to have "creating, changing, and safeguarding passwords". Kameleoon and VWO state this means you must force users to update their passwords every 90 days, but this is based on [out of date advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) first published in 2004. [Updated advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) published by the National Institute of Standards and Technology in 2017 states "Verifiers SHOULD NOT require memorized secrets [passwords] to be changed arbitrarily (e.g., periodically)."  

Author: Think I'm going to move this into an FAQ at the bottom of the article and boxout feels weird, but I don't want it in the intro section either.

#### Adobe Target and HIPAA compliance

- **Self-Hosting:** No
- **BAA Available**: Yes

What [Adobe says](https://www.adobe.com/trust/compliance/hipaa-ready.html):

> "Adobe provides health care customers with services that are ready to accept PHI, referring to these services as HIPAA-Ready Services. These HIPAA-Ready Services have additional features and functionalities that allow for both customers, who are Covered Entities or Business Associates, and Adobe to comply with their respective HIPAA obligations. These additional features may increase your license or subscription costs."

#### How much does Adobe Target cost?

Adobe doesn't publish pricing for Adobe Target, but suffice it to say it probably won't be cheap given its typical clientele are multinational corporations. As noted in its T&Cs, using its "HIPAA-Ready Service" may incur an additional cost above a normal subscription plan.

## FAQ

### What are the password requirements under HIPAA?

> HIPAA requires Covered Entities and Business Associates to have "creating, changing, and safeguarding passwords". Kameleoon and VWO state this means you must force users to update their passwords every 90 days, but this is based on [out of date advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) first published in 2004. [Updated advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) published by the National Institute of Standards and Technology in 2017 states "Verifiers SHOULD NOT require memorized secrets [passwords] to be changed arbitrarily (e.g., periodically)."

Author: ```suggestion
HIPAA requires Covered Entities and Business Associates to have policies for "creating, changing, and safeguarding passwords". Kameleoon and VWO state this means you must force users to update their passwords every 90 days, but this is based on [out of date advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) first published in 2004. [Updated advice](https://www.hipaajournal.com/what-are-the-hipaa-password-expiration-requirements/) published by the National Institute of Standards and Technology in 2017 states "Verifiers SHOULD NOT require memorized secrets [passwords] to be changed arbitrarily (e.g., periodically)."
```

Why? It's logical, really. Research shows that, when required to change passwords frequently, users are more likely to use weaker, easy to remember passwords and make small, predictable changes than use a strong one. Anyone who's had to change their password every 30 days knows this is true already.

### What's the difference between A/B and multivariate testing?

**A/B testing** involves comparing the conversion rates of two or more different versions of the same page or user experience. Users are sent to one version or the other, and the outcomes are compared until a large enough sample has been achieved to measure the statistical significance of the results. A/B testing is sometimes called 'split testing' and you can test more than two pages, a process sometimes referred to as A/B/n testing.

> **Multivariate testing** is similar to A/B testing, but it tests more page elements together to understand how they interact with each other. One would, for example, use a multivariate test to compare all combinations of three different page headlines and three different CTA (call to action) when paired together. The basic process is similar to an A/B test, but instead of comparing whole page designs, your're comparing specific elements to each other.

Author: ```suggestion
**Multivariate testing** is similar to A/B testing, but it tests more page elements together to understand how they interact with each other. One would, for example, use a multivariate test to compare all the possible combinations of three different page headlines and three different CTAs (call to actions) to see which performs best. The basic process is similar to an A/B test, but instead of comparing whole page designs, your're comparing specific elements to each other.
```
